File: fabric/src/main/java/dev/wren/truss/platform/FabricPlatformHelper.java
package dev.wren.truss.platform;

import dev.wren.truss.platform.services.PlatformHelper;
import net.fabricmc.api.EnvType;
import net.fabricmc.loader.api.FabricLoader;

public class FabricPlatformHelper implements PlatformHelper {
    @Override
    public boolean isClient() {
        return FabricLoader.getInstance().getEnvironmentType() == EnvType.CLIENT;
    }

    @Override
    public boolean isDedicatedServer() {
        return FabricLoader.getInstance().getEnvironmentType() == EnvType.SERVER;
    }

    @Override
    public String getDist() {
        return FabricLoader.getInstance().getEnvironmentType().name();
    }

    @Override
    public Platform getPlatform() {
        return Platform.FABRIC;
    }
}

File: fabric/src/main/kotlin/dev/wren/truss/platform/FabricPlatformHelper.kt
package dev.wren.truss.platform

import dev.wren.truss.platform.services.PlatformHelper
import net.fabricmc.api.EnvType
import net.fabricmc.loader.api.FabricLoader

class FabricPlatformHelper : PlatformHelper {
    override val isClient: Boolean = FabricLoader.getInstance().environmentType == EnvType.CLIENT
    override val isDedicatedServer: Boolean = FabricLoader.getInstance().environmentType == EnvType.SERVER
    override val dist: String = FabricLoader.getInstance().environmentType.name
    override val platform: PlatformHelper.Platform = PlatformHelper.Platform.FABRIC
}